packages/liveblocks-core/src/comments/index.ts

```diff
@@ -58,6 +59,16 @@ export type CommentsApi<TThreadMetadata extends BaseMetadata> = {
   }): Promise<CommentData>;
 };
 
+export class CommentsApiError extends Error {
+  constructor(
+    public message: string,
+    public status: number,
+    public details?: JsonObject
+  ) {
+    super(message);
+  }
+}
```

Comment:
Not a super big fan of using custom error classes in JS, because they add complexity to error handlers elsewhere and are hard to maintain over time. At the same time, I also don't have a great better suggestion to use right now. But maybe I'll want to unify the error handling across the board at one point in the future.

At the very least, can we make sure such custom error class is a private API and this class won't show up in the `*.d.ts` files, so we're free to refactor this later without users relying on this implementation detail? You can ensure that by adding an `/** @internal */` JSDoc comment.